data/moves.ts

```diff
@@ -14922,6 +14922,7 @@ export const Moves: import('../sim/dex-moves').MoveDataTable = {
 								if (action.choice === 'megaEvo') this.actions.runMegaEvo(source);
 								// Also a "forme" change that happens before moves, though only possible in NatDex
 								if (action.choice === 'terastallize') this.actions.terastallize(source);
+								else continue;
```

Comment:
This still isn't quite right because it doesn't cancel the `megaEvo` action.
```suggestion
								if (action.choice === 'megaEvo') this.actions.runMegaEvo(source);
								// Also a "forme" change that happens before moves, though only possible in NatDex
								else if (action.choice === 'terastallize') this.actions.terastallize(source);
								else continue;
```

Reply:
> This still isn't quite right because it doesn't cancel the `megaEvo` action.

thanks! didn't notice that [edd2b8e](https://github.com/smogon/pokemon-showdown/pull/10624/commits/edd2b8ecf7453e151bfad74d457e87479175e716) didn't have equivalent behavior to the original switch I added.